### PR TITLE
Add support for Grape based APIs

### PR DESCRIPTION
**Pull request by tlconnor**
This PR adds support for API endpoints defined using Grape (https://github.com/ruby-grape/grape)

### CODE REVIEW

File: lib/declarative_authorization/controller/grape.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            action_name = "#{request.request_method} #{route.origin}"
+
+            permissions = api_class.all_filter_access_permissions
+            all_permissions = permissions.select {|p| p.actions.include?(:all)}
+            matching_permissions = permissions.select {|p| p.matches?(action_name)}
+            allowed = false
+            auth_exception = nil
+
+            begin
+              allowed = if !matching_permissions.empty?
```
Comment (mathewbaltes) [inline lib/declarative_authorization/controller/grape.rb]: Could we use `.present?` instead

File: lib/declarative_authorization/controller/grape.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                          matching_permissions.all? {|perm|
+                            perm.permit!(self, action_name)
+                        }
+                        elsif !all_permissions.empty?
```
Comment (mathewbaltes) [inline lib/declarative_authorization/controller/grape.rb]: Same here.

File: lib/declarative_authorization/controller/grape.rb (hunk 1 above)
Comment (dvicory) [reply lib/declarative_authorization/controller/grape.rb]: If we do then it should be changed in both the Rails and Grape versions in a new commit (code was duplicated). 😄

Comment (tlconnor) [reply lib/declarative_authorization/controller/grape.rb]: This was existing code taken from `lib/declarative_authorization/in_controller.rb` that also exists in `lib/declarative_authorization/controller/rails.rb`.

For the sake of keeping this PR smaller I chose not to clean up existing code.

Comment (tlconnor) [reply lib/declarative_authorization/controller/grape.rb]: @dvicory good point about code duplication. Perhaps further refactoring is needed.

**Pull request merged**